mitreattack/attackToExcel/stixToDf.py

```diff
@@ -383,67 +516,19 @@ def matricesToDf(src, domain):
-                    for i in range(len(subtechniques)):  # for each sub-technique
-                        if i != 0: techniques_column.append(
-                            "")  # first sub-technique is parallel to the technique in the layout
-                        subtechniques_column.append(subtechniques[i]["name"])
-                    technique_bottom = len(techniques_column) + 1  # bottom of row range to merge
-                    if technique_top != technique_bottom:  # more than 1 sub-technique
-                        merge.append(CellRange(  # merge technique portion of cell group
-                            len(columns),
-                            len(columns),
-                            technique_top,
-                            technique_bottom,
-                            data=technique["name"],
-                            format={  # format of the merged range
-                                "name": "supertechnique",
-                                "format": {
-                                    'valign': 'vcenter',
-                                    'text_wrap': 1,
-                                    'shrink': 1
-                                }
-                            }
-                        ))
-                else:  # no sub-techniques; add empty cell parallel to technique
-                    subtechniques_column.append("")
-            # end adding techniques and sub-techniques to column
-
-            matrix_grid.append(techniques_column)  # add technique column to grid
-            if len(list(filter(lambda x: x != "",
-                               subtechniques_column))) > 0:  # if there are sub-techniques for the tactic as well
-                matrix_grid.append(subtechniques_column)  # add sub-technique sub-column
-                columns.append("")  # add empty tactic header for the sub-column
-                merge.append(  # merge tactic column header with the sub-column header that was just appended
-                    CellRange(len(columns) - 1,
-                              len(columns),
-                              1,
-                              1,
-                              data=tactic["name"],
-                              format={  # tactic header formatting
-                                  "name": "tacticHeader",
-                                  "format": {
-                                      "bold": 1,
-                                      "border": 1,
-                                      "font_size": 14,
-                                      "align": "center",
-                                      "shrink": 1
-                                  }
-                              }
-                              ))
+            build_technique_and_sub_columns(src, techniques, columns, merge, matrix_grid, tactic['name'])
+
+            for platform in Matrix_Platforms_Lookup[domain]:
```

Comment:
Given how many matrices there are we may want to wrap this in a `tqdm` loading bar since it appears to freeze while it's building matrices. If added we should remove the `building matrices... done` text from `matricesToDf`.

Reply:
Turns out this is a little harder than I would have initially liked, but I've got something that shows progress. Let me know if you're not a fan of this particular implementation/visual, and I can try go make something better.